service/r2/routes.go

```diff
@@ -271,3 +280,116 @@ func fetchRollupRuleHistory(s *service, r *http.Request) (data interface{}, err
 	}
 	return models.NewRollupRuleSnapshots(hist), nil
 }
+
+func updateRuleSet(s *service, r *http.Request) (data interface{}, err error) {
+	vars := mux.Vars(r)
+	var req updateRuleSetRequest
+	if err := parseRequest(&req, r.Body); err != nil {
+		return nil, err
+	}
+	uOpts, err := s.newUpdateOptions(r)
+	if err != nil {
+		return nil, err
+	}
+
+	originalRS, err := s.store.FetchRuleSet(vars[namespaceIDVar])
+	if err != nil {
+		return nil, err
+	}
+	if req.RuleSetVersion != originalRS.Version() {
+		return nil, NewConflictError(
+			fmt.Sprintf("can not update namespace %s, version mismatch", vars[namespaceIDVar]),
+		)
+	}
+
+	newRS, err := applyChangesToRuleSet(
```

Comment:
The convention here for all the handlers is to implement the logic that mutates ruleset in`m3metrics/rules/ruleset.go`, and call the corresponding function here. Can you please do that? You can use `UpdateMappingRule` as an example to see the pattern here.